xonsh/shell.py

```diff
@@ -75,6 +75,15 @@
 """,
 )
 
+events.doc(
+    "on_before_prompter_prompt",
```

Comment:
probably should follow the pre/post naming convention, ie `on_pre_prompter_prompt`